### ISSUE TITLE
[BUG] k2a on chromebook does not work

### ISSUE THREAD

**Issue by MrSellerie** [PhilippC/keepass2android#2520]
### Checks

- [X] I have read the FAQ section, searched the open issues, and still think this is a new bug.

### Describe the bug you encountered:

New problem on Google_Vilboz.13434.576.0 Zork and Android 13 in Chomebook-Playstore. App starts, seems to be running for a few seconds and the app ends without an error message. After deleting cache and/or data you can attach to google drive, but after a few seconds ends the app also without a message.

[edit]
Seems to be a permission problem. Camera and contacts can be enabled but i could not give access to the file system.

[edit2]
App de-inst and re-inst after a needed chromebook reboot for v120. No change, app ended without an error message. Version 1.10-PRE did not ask for file access on 2 devices.
Can somebody check the needed permissions on your devices?

Google Chrome: Version 120.0.6099.235 (Official Build) (64-bit)
Platform: 15662.76.0 (Official Build) stable-channel zork
Chrome ARC Android SDK Version: 33
ChromeOS Firmware Version: Google_Vilboz.13434.576.0
ARC Enabled: true
ARC: 11277367
Channel: stable-channel
Enterprise Enrolled: false
Developer Mode: false

### What version of Keepass2Android are you using?

1.10-PRE

### Which version of Android are you on?

13

**Comment by Spiralpine** [comment]
The App info for KP2A on my Chromebook lists the permissions as follows:
- Camera = Not allowed
- Contact = Allowed
- Storage = Allowed

All are the default permissions the app asks you during the installation. 

Also, My Chromebook is running ChromeOS  with this detail:

Google Chrome: Version 120.0.6099.235 (Official Build)  (64-bit)
Platform: 15662.76.0 (Official Build) stable-channel eve
Channel: stable-channel
Firmware Version: Google_Eve.9584.230.0
ARC Enabled: true
ARC: 11277394
Enterprise Enrolled: false
Developer Mode: false

Also, under "Settings\Apps\Google Play Store\Managed Android Preferences\About Device\Android version", it shows it's running Android 11.  Are you sure you are running Android 13? Where did you get that you are running Android 13?

**Comment by MrSellerie** [comment]
Yep, "Settings\Apps\Google Play Store\Managed Android Preferences\About Device\Android version" displays 13.

**Comment by MrSellerie** [comment]
Android 13

**Comment by Spiralpine** [comment]
In my case, on a new install or even after clearing the cache and storage, when I choose Goggle Drive, the app always crashes with no error message. I always need to launch the app again to make it work and it works afterward with no issue. 

Allow those permissions, which I mentioned above,  separately and outside of the app  (since you are saying those permissions are not asked in your case) and launch the app more than once to see what happens.

**Comment by MrSellerie** [comment]
I cleared cache and storage and re-installed the app more than one times. An app relaunch does not help, after choosing Google Drive (has to be done on every app start), the app always crashes without an error message. It makes also no difference, if i choose the entire Google Drive or only the part for KP2A.

The permission for storage is grayed out. After clearing cache and storage or app re-inst, the app asks the permissions only for camera and contacts.

@Spiralpine 
Do you work with a local copy of the KP2A database?

**Comment by Spiralpine** [comment]
I do not use the local copy of the DB. Only the DB residing in the Google drive folder.

Here is what my permissions look like:

![image](https://github.com/PhilippC/keepass2android/assets/55951150/119d74f7-047a-4125-b248-2754ed777991)

I would suggest you try these:

1 - Try to open a copy of your DB saved locally to your download folder (open it using the system file picker) and see the outcome.  This would try to eliminate any GD issues.
2 - Create a new test dummy DB with a dummy entry saved locally and on your GD and try to open each to see the results. This would aim to eliminate any issue with your main DB itself.
3 - If #1 and #2 are not conclusive, then generate a sanitized KP2A log and attach it here for the developer to look at.

**Comment by Coasie** [comment]
I see this same force/close without error.

I have 2 new late 2023 chromebooks one is an Acer Spin 714 with an Intel I5, and sideloaded K2A 1.10-pre works just fine.
The other is an Asus 14' with an AMD Ryzen 3 7320c and sideloaded K2A 1.10-pre crashes when opening.

Both run the latest ChroemOS 120, and I have also tried with 118 and 119 when they were the latest version. The Asus/AMD model, K2A always crashes when loading.

**Comment by MrSellerie** [comment]
ChromeOS 122 and 123 changed nothing.

**Comment by MrSellerie** [comment]
After deleting app and cache data, the app in version 1.10-pre can be opened again. I can only set permissions for Camera, Contacts and Notifications. Added a local file as database with a short password, hitting the create button and the app closes itself after a few seconds.
Re-installing the app does not change the app version. still 1.10-pre.

**Comment by Coasie** [comment]
Yup, it opens, but you can't link to a cloud file, the app just force closes.

**Comment by MrSellerie** [comment]
4 months later without progress. Next chromebook and et voila the same problem. The app opens, closes after a few seconds and no message is displayed.

> Google Chrome	126.0.6478.222 (Offizieller Build) (64-Bit) 
Überarbeitung	803933f6b3177592370be766f36d7095a1903fd0-refs/branch-heads/6478@{#1853}
Plattform	15886.74.0 (Official Build) stable-channel skyrim
Firmwareversion	Google_Markarth.15390.255.0
Personalisierungs-ID	markarth
ARC	12058831 SDK-Version: 33
JavaScript	V8 12.6.228.28
User-Agent	Mozilla/5.0 (X11; CrOS x86_64 14541.0.0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36
Android	V13

**Comment by MrSellerie** [comment]
Update. The app runs without crashing as long as you do not add a file. Tried with Android-Filebrowser and both Google Drive entries.

**Comment by MrSellerie** [comment]
problem still exists since months without a sign of life from the developer...

**Comment by MrSellerie** [comment]
KeepassDX is not a permanent solution with ChromeOS v129/v130 and Android v13. KBDX files can be opened locally and externally without the app crashing. Good, but you cannot write back changes to external files directly. Changes can only be saved via ‘Save as’. However, this leads to a new unique file reference in Google Drive. Deleting the old file and renaming the new one does not change anything. The newly created file must be selected again on all other devices, even if the same file name is used. File name and date are apparently only managed as object properties for a reference and not all Keepass implementations use the pure file name.

**Comment by MrSellerie** [comment]
Problem still exists.
